### PR TITLE
LogContext: Add header and close button to modal

### LINKED ISSUE #56282: Modal should be titled 'Show Context' so the users understands the purpose of the modal.

### PR DESCRIPTION
**Pull request by svennergr**
**What this PR does / why we need it**:
This PR adds a new header to the show-context modal containing a button to close the header.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/56282, https://github.com/grafana/grafana/issues/56281

**Special notes for your reviewer**:

image

### CODE REVIEW

File: public/app/features/logs/components/LogRowContext.tsx [hunk 1]
```diff
@@ -61,26 +65,60 @@ const getLogRowContextStyles = (theme: GrafanaTheme2, wrapLogMessage?: boolean)
...
+    title: css`
+      position: absolute;
+      top: -${contextHeight + headerHeight}px;
+      z-index: ${theme.zIndex.modal};
+      height: ${headerHeight}px;
       background: ${theme.colors.background.secondary};
+      border: 1px solid ${theme.colors.background.secondary};
+      border-radius: ${theme.shape.borderRadius(2)} ${theme.shape.borderRadius(2)} 0 0;
+      box-shadow: 0 0 10px ${theme.v1.palette.black};
+      clip-path: inset(-10px -10px 0px -10px);
```
Comment (svennergr) [inline public/app/features/logs/components/LogRowContext.tsx:91]: I am using box-shadows in combination with `clip-path` to hide shadows on the relevant sides.

File: public/app/features/logs/components/LogRowContext.tsx [hunk 2]
```diff
@@ -261,6 +301,12 @@ export const LogRowContext: React.FunctionComponent<LogRowContextProps> = ({
             logsSortOrder={logsSortOrder}
           />
         )}
+        <div className={cx(title, sizes)}>
+          <h5>Show Context</h5>
```
Comment (ivanahuckova) [inline public/app/features/logs/components/LogRowContext.tsx:305]:
```suggestion
          <h5>Log context</h5>
```
Maybe Log context?

**Review by ivanahuckova — APPROVED** [review]
LGTM as hot fix, but we should refactor this to use modal instead of custom overrides and styling. But I think we are all on the same page regarding this.

File: public/app/features/logs/components/LogRowContext.tsx (hunk 2 above)
Comment (svennergr) [reply public/app/features/logs/components/LogRowContext.tsx]: Good Idea! Renamed in https://github.com/grafana/grafana/pull/56283/commits/519a1ecbc56ffe5a90d50eadbee334b44a5b3e2b

**Pull request merged**